Homework/2/Program.cs

```diff
@@ -0,0 +1,470 @@
+
+            for (int i = 0; i < userArray.Length; i++)
+            {
+                Console.WriteLine(userArray[i][0]);
+            }
+            PromptInput("\nPress any key to continue.");
+            MenuSelection();
+        }
+
+        //Function to add userDetails to end or start of userArray. Taking bool variables to determine whether add to start or end or select index.
+        static void PromptAddUserToArray(bool addToEnd = true,bool addToStart = false) {
+            Console.Clear();
+            Console.WriteLine("Adding user:\n");
+
+            string username = PromptInput("Enter username:");
+            string password = PromptInput("Enter password:");
+
+
+            int index = GetIndex(addToEnd, addToStart);
+
+            if (!CheckIfUserExists(username) && !(password == "") && index != -1)
+            {
+                string[] userDetails = new string[2] { username, password };
+                AddUserToArrayAtIndex(userDetails, index);
+                WriteUsersToFile();
+                ListUsers();
+            }
+            else {
+                if (RetryConfirmation("Bad user details."))
+                {
+                    PromptAddUserToArray(addToEnd, addToStart);
+                }
+                else
+                {
+                    MenuSelection();
+                }
+            }
+        }
+
+        //Function to get index of where user should be removed
+        static void PromptRemoveUser(bool removeFromEnd = true, bool removeFromStart = false) {
+            Console.Clear();
+            int index = GetIndex(removeFromEnd, removeFromStart);
+            if (index!=-1)
+            {
+                RemoveUser(index);
+                WriteUsersToFile();
+            }
+            ListUsers();
+        }
+
+        //returns or prompts index for array. Returns -1 if there are issues.
+        static int GetIndex(bool endOfArray = true, bool startOfArray = false) {
+            if (endOfArray)
+            {
+                return userArray.Length;
+            }
+            else if (startOfArray)
+            {
+                return 0;
+            }
+            else if (!endOfArray && !startOfArray)
+            {
+                string indexStr = PromptInput("Enter index:");
+                int.TryParse(indexStr, out int index);
+
+                if (index < 0 || index >= userArray.Length)
+                {
+                    return -1;
+                }
+                else {
+                    return index;
+                }
+            }
+            return -1;
+        }
+
+
+        //Remove user from userArray at index.
+        static void RemoveUser(int index) {
+            string[][] newArray = new string[userArray.Length - 1][];
+
+            for (int i = 0; i < newArray.Length; i++)
+            {
+                if (i <index)
+                {
+                    newArray[i] = userArray[i];
+                }else if (i == index)
```

Comment:
This block and the next block are identical.  Seems you could simplify.

Reply:
Removed unneeded else if.